geonode/geoserver/security.py

```diff
@@ -32,136 +32,27 @@
-        allowed_area.text = geo_limit
-    else:
-        access_el = etree.SubElement(root_el, "access")
-        access_el.text = access
-    return etree.tostring(root_el)
-
-
-def _update_geofence_rule(layer, dataset_name, workspace,
-                          service, request=None,
-                          user=None, group=None,
-                          geo_limit=None, allow=True):
-    payload = _get_geofence_payload(
-        layer=layer,
-        dataset_name=dataset_name,
-        workspace=workspace,
-        access="ALLOW" if allow else "DENY",
-        user=user,
-        group=group,
-        service=service,
-        request=request,
-        geo_limit=geo_limit
-    )
-    logger.debug(f"request data: {payload}")
-    response = requests.post(
-        f"{settings.OGC_SERVER['default']['LOCATION']}rest/geofence/rules",
-        data=payload,
-        headers={
-            'Content-type': 'application/xml'
-        },
-        auth=HTTPBasicAuth(
-            username=settings.OGC_SERVER['default']['USER'],
-            password=settings.OGC_SERVER['default']['PASSWORD']
-        )
-    )
-    logger.debug(f"response status_code: {response.status_code}")
-    if response.status_code not in (200, 201):
-        msg = (f"Could not ADD GeoServer User {user} Rule for "
-               f"Dataset {layer}: '{response.text}'")
-        if 'Duplicate Rule' in response.text:
-            logger.debug(msg)
-        else:
-            raise RuntimeError(msg)
-
-
-def get_geofence_rules(page=0, entries=1, count=False):
-    """Get the number of available GeoFence Cache Rules"""
-    try:
-        url = settings.OGC_SERVER['default']['LOCATION']
-        user = settings.OGC_SERVER['default']['USER']
-        passwd = settings.OGC_SERVER['default']['PASSWORD']
-
-        _url = ''
-        _headers = {'Content-type': 'application/json'}
-        if count:
-            """
-            curl -X GET -u admin:geoserver \
-                http://<host>:<port>/geoserver/rest/geofence/rules/count.json
-            """
-            _url = f"{url}rest/geofence/rules/count.json"
-        elif page or entries:
-            """
-            curl -X GET -u admin:geoserver \
-                http://<host>:<port>/geoserver/rest/geofence/rules.json?page={page}&entries={entries}
-            """
-            _url = f'{url}rest/geofence/rules.json?page={page}&entries={entries}'
-        r = requests.get(_url,
-                         headers=_headers,
-                         auth=HTTPBasicAuth(user, passwd),
-                         timeout=10,
-                         verify=False)
-        if (r.status_code < 200 or r.status_code > 201):
-            logger.debug("Could not retrieve GeoFence Rules count.")
-
-        rules_objs = json.loads(r.text)
-        return rules_objs
-    except Exception:
-        tb = traceback.format_exc()
-        logger.debug(tb)
-        return {'count': -1}
-
+def create_geofence_client():
```

Comment:
I would create an instance of the `client` on the `geoserver.helpers` module, similarly on how we do already for the `catalog` and `importer` ones.